packages/mdc-floating-label/README.md

```diff
@@ -0,0 +1,106 @@
+<!--docs:
+title: "Floating Label"
+layout: detail
+section: components
+excerpt: "An animated text caption for a text field or select."
+path: /catalog/input-controls/floating-label/
+-->
+
+# Floating Label
+
+Floating labels display the type of input a field requires. Every text field and select should have a label. Labels are aligned with the input line and always visible. They can be resting (when a field is inactive and empty) or floating. The label is a text caption or description for the text field.
+
+## Design & API Documentation
+
+<ul class="icon-list">
+  <li class="icon-list-item icon-list-item--spec">
+    <a href="https://material.io/guidelines/components/text-fields.html#text-fields-layout">Material Design guidelines: Text Fields Layout</a>
+  </li>
+</ul>
+
+## Usage
+
+### HTML Structure
+
+```html
+<label class="mdc-floating-label" for="my-text-field-id">Hint text</label>
+```
+
+### Usage within `mdc-text-field`
+
+```html
+<div class="mdc-text-field">
+  <input type="text" id="my-text-field-id" class="mdc-text-field__input">
+  <label class="mdc-floating-label" for="my-text-field-id">Hint text</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+```
+
+<!-- TODO(mattgoo): add ### Usage within `mdc-select` once select uses mdc-floating-label -->
+
+#### Avoid Dynamic ID Generation
+
+It's also possible to wrap `mdc-text-field__input` within a `<label>` to avoid dynamic id generation:
+
+```html
+<label class="mdc-text-field">
+  <input type="text" class="mdc-text-field__input">
+  <span class="mdc-floating-label">Hint Text</span>
+  <div class="mdc-text-field__bottom-line"></div>
+</label>
+```
+
+> _NOTE_: Only place a `mdc-floating-label` inside of a text field _if you plan on using
```

Comment:
Didn't we drop support for CSS-only text fields and selects?

If so, should we delete this paragraph and the "CSS Only" section below it?